src/Testcontainers/Clients/DockerImageOperations.cs

```diff
@@ -93,6 +103,14 @@ public async Task<string> BuildAsync(IImageFromDockerfileConfiguration configura
       var image = configuration.Image;
 
       ITarArchive dockerfileArchive = new DockerfileArchive(configuration.DockerfileDirectory, configuration.Dockerfile, image, _logger);
+      var images = await GetDependentImagesAsync( configuration, ct );
```

Comment:
I think we can simplify a lot here. Considering Java's regex to extract the image, I recommend something like:

```csharp
private static readonly Regex FromLinePattern = new Regex("FROM (?<arg>--[^\\s]+\\s)*(?<image>[^\\s]+).*", RegexOptions.None, TimeSpan.FromSeconds(5));

_ = File.ReadAllLines(dockerfileFilePath)
    .Select(line => FromLinePattern.Match(line))
    .Where(match => match.Success)
    .Select(match => match.Groups["image"])
    .Select(group => group.Value);
```